providers/provider_data.go

```diff
@@ -168,6 +172,30 @@ func (p *ProviderData) buildSessionFromClaims(idToken *oidc.IDToken) (*sessions.
 		return nil, fmt.Errorf("email in id_token (%s) isn't verified", claims.Email)
 	}
 
+	if p.SkipAudCheckWhenMissing {
```

Comment:
These token -> session conversions in bearer auth are covered here: https://github.com/oauth2-proxy/oauth2-proxy/blob/06808704a3762de73bed1cc92b077420eb93d735/oauthproxy.go#L310

The challenge is we use `go-oidc` library to verify OIDC IDTokens. That library will not validate tokens without a matching `aud` if I recall.

I don't think this gets around that constraint at all, right?

Reply:
You need to use `SkipClientIDCheck` on the underlying OIDC Verifier if I recall.